File: java-streams/example/src/main/java/streams/Terminal.java
package streams;

import java.util.List;
import java.util.OptionalDouble;
import java.util.stream.Collectors;
import java.util.stream.IntStream;
import java.util.stream.Stream;

public class Terminal {

    void examples(List<Person> persons) {

        Stream<Person> stream = persons.stream();

        String lastNames = stream
                .map(person -> person.getLastName())
                .collect(Collectors.joining());

        List<String> lastNamesList = stream
                .map(Person::getLastName)
                .collect(Collectors.toList());

        stream.forEach(person -> System.out.println("person.getLastName() = " + person.getLastName()));

        IntStream intStream = IntStream.range(0, 20);
        long count = intStream.count();
        int sum = intStream.sum();
        OptionalDouble average = intStream.average();
    }
}

File: java-streams/examples/kotlin/src/main/kotlin/streams/Terminal.kt
package streams

class Terminal {

    internal fun examples(persons: List<Person>) {

        val stream = persons

        val lastNames = stream
                .map { it.lastName }
                .joinToString()

        val lastNamesList = stream
                .map { it.lastName }
                .toList()

        stream.forEach { (_, _, lastName) -> println("person.getLastName() = $lastName") }

        val intStream = 0..20
        val count = intStream.count()
        val sum = intStream.sum()
        val average = intStream.average()
    }
}
